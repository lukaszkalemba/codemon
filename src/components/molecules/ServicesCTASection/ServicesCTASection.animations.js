const getWrapper = animate => {
  return {
    variants: {
      animate: {
        transition: {
          staggerChildren: 0.2,
        },
      },
    },
    animate,
    initial: 'initial',
  };
};

const children = {
  variants: {
    animate: {
      opacity: 1,
      y: 0,
      transition: { duration: 0.8 },
    },
    initial: { opacity: 0, y: 50 },
  },
};

export default { getWrapper, children };
