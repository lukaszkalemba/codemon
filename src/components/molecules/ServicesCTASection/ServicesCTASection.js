import React from 'react';
import PropTypes from 'prop-types';
import { motion } from 'framer-motion';
import Container from 'components/particles/Container';
import Link from 'components/atoms/Link';
import S from './ServicesCTASection.styles';
import {
  wrapperVariants,
  childrenVariants,
} from './ServicesCTASection.animations';

const ServicesCTASection = ({ animation }) => {
  return (
    <Container axis="both">
      <motion.div
        animate={animation}
        initial="initial"
        variants={wrapperVariants}
      >
        <S.Heading variants={childrenVariants}>
          Stwórz z nami od zera niepowtarzalny projekt, lub przenieś już
          istniejący na kolejny poziom.
        </S.Heading>
        <motion.div variants={childrenVariants}>
          <Link to="/wycena">Bezpłatna wycena</Link>
        </motion.div>
      </motion.div>
    </Container>
  );
};

ServicesCTASection.defaultProps = {
  animation: {},
};

ServicesCTASection.propTypes = {
  animation: PropTypes.shape({}),
};

export default ServicesCTASection;
