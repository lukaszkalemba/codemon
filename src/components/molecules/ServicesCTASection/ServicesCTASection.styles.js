import styled from 'styled-components';
import { motion } from 'framer-motion';

const Heading = styled(motion.h2)`
  font-size: ${({ theme }) => theme.font.size.s};
  line-height: 1.75em;
  max-width: 70vw;
  margin-top: 0;

  ${({ theme }) => theme.media.tablet`
    font-size: ${theme.font.size.m};
  `}

  ${({ theme }) => theme.media.laptop`
    max-width: 600px;
    margin-top: 200px;
  `}

  ${({ theme }) => theme.media.desktop`
    font-size: ${theme.font.size.l};
  `}
`;

export default { Heading };
