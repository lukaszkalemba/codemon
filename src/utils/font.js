export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
