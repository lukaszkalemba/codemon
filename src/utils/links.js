export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
