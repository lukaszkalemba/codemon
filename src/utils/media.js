export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
