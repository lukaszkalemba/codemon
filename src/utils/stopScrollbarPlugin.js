import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
