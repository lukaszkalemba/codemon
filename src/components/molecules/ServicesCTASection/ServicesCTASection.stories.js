import React from 'react';
import ServicesCTASection from './ServicesCTASection';

export default {
  title: 'Molecules/ServicesCTASection',
  component: ServicesCTASection,
};

const Template = args => <ServicesCTASection {...args} />;

export const Default = Template.bind({});
Default.args = {};
