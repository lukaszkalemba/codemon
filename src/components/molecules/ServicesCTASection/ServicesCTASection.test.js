import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import ServicesCTASection from './ServicesCTASection';

describe('<ServicesCTASection />', () => {
  it('renders correctly and matches snapshot', () => {
    const { container } = render(<ServicesCTASection />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
