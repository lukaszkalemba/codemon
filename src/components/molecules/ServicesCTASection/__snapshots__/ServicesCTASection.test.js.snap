// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServicesCTASection /> matches snapshot 1`] = `
<div
  class="sc-bdnylx bZvVzr"
>
  <div>
    <h2
      class="sc-jSFkmK kNfYpP"
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      Stwórzmy od zera niepowtarzalny projekt, lub przenieśmy już istniejący na kolejny poziom.
    </h2>
    <div
      style="opacity: 0; transform: translateY(50px) translateZ(0);"
    >
      <div
        class="sc-dlnjPT ka-Dptu"
      >
        <a
          class="sc-hKFyIo fGlMDG"
          href="/wycena"
        >
          Bezpłatna wycena
          <div
            class="sc-eCApGN drKECk"
          >
            <div
              class="sc-gtssRu fiyWPK"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 65 16"
                width="65"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1H61L42.5 14.5"
                  stroke="white"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
